src/rpm_ostree/cli_status.rs

```diff
@@ -98,12 +98,23 @@ pub fn parse_booted(status: &StatusJson) -> Result<Release> {
 }
 
 /// Parse updates stream for booted deployment from status object.
-pub fn parse_updates_stream(status: &StatusJson) -> Result<String> {
+pub fn parse_booted_updates_stream(status: &StatusJson) -> Result<String> {
     let json = booted_json(status)?;
     ensure!(!json.base_metadata.stream.is_empty(), "empty stream value");
     Ok(json.base_metadata.stream)
 }
 
+/// Parse updates stream for pending deployment from status object.
+pub fn parse_pending_updates_stream(status: &StatusJson) -> Result<String> {
+    let pending_json = status.deployments[0].clone();
```

Comment:
Does this implicitly expect the first deployment to be the pending one? It does seem a bit fragile (and possibly wrong).

Reply:
Yes, this does expect the first deployment to be the pending one.

Reply:
It would be good not to hardcode that assumption, we can pick up the staged/pending deployment through properties.